web_api/core/views.py

```diff
@@ -229,23 +238,29 @@ def update_subscription(request: HttpRequest, team_id: str) -> HttpResponse:
         items=[
             {
                 "id": subscription["items"]["data"][0].id,
-                "plan": stripe_customer_info.plan_id,
-                "quantity": seats,
+                "plan": plan_id_from_period(period=payload.planPeriod),
+                "quantity": payload.seats,
             }
         ],
-        proration_date=proration_timestamp,
-    )
-    # when we upgrade a users plan Stripe will charge them on their next billing
-    # cycle. To make Stripe charge for the upgrade immediately we must create an
-    # invoice and pay it. If we don't pay the invoice Stripe will wait 1 hour
-    # before attempting to charge user.
-    invoice = stripe.Invoice.create(
-        customer=stripe_customer_info.customer_id, auto_advance=True
+        proration_date=payload.prorationTimestamp,
     )
-    # we must specify the payment method because our Stripe customers don't have
-    # a default payment method, so the default invoicing will fail.
-    stripe.Invoice.pay(invoice.id, payment_method=subscription.default_payment_method)
+    # we only need to manually created invoices when a user modifies their
+    # subscription within the same billing period.
+    if payload.planPeriod == subscription.plan.interval:
```

Comment:
This is super weird. I don't really understand it, but it's the reality. Users get charged immediately when changing between plans but not when upgrading.